Add Gemini Embedding Model 001 (#457)
qdrant/landing_page: qdrant-landing/content/documentation/embedding/gemini.md

---
title: Gemini
weight: 700
---

# Gemini

Qdrant is compatible with Gemini Embedding Model API and its official Python SDK that might be installed as any other package:

Gemini is a new family of Google PaLM models, released in December 2023. The new embedding models succeed the previous Gecko Embedding Model. 

> The new models allow passing an additional parameter to the API call: `task_type`. It determines the type of task you want to use the embeddings for.

Reviewer: ```suggestion
In the latest models, an additional parameter, task_type, can be passed to the API call. This parameter serves to designate the intended purpose for the embeddings utilized.

The Embedding Model API supports various task types, outlined as follows:
```

> These are the task types that are supported by the Embedding Model API:

Reviewer: ```suggestion

```

> - TASK_TYPE_UNSPECIFIED (0): Unset value, which will default to one of the other enum values.

Reviewer: ```suggestion
- task_type_unspecified: Unset value, which will default to one of the other values.
```

> - RETRIEVAL_QUERY (1): Specifies the given text is a query in a search/retrieval setting.

Reviewer: ```suggestion
- retreival_query: Specifies the given text is a query in a search/retrieval setting.
```

> - RETRIEVAL_DOCUMENT (2): Specifies the given text is a document from the corpus being searched.

Reviewer: ```suggestion
- retrieval_document: Specifies the given text is a document from the corpus being searched.
```

> - SEMANTIC_SIMILARITY (3): Specifies the given text will be used for STS.

Reviewer: ```suggestion
- semantic_similarity: Specifies the given text will be used for Semantic Text Similarity.
```

> - CLASSIFICATION (4): Specifies that the given text will be classified.

Reviewer: ```suggestion
- classification: Specifies that the given text will be classified.
```

> - CLUSTERING (5): Specifies that the embeddings will be used for clustering.

Reviewer: ```suggestion
- clustering: Specifies that the embeddings will be used for clustering.
```

> 

Reviewer: I suggest we lowercase the values as used with parameter "task_type" as used below.
